### PR TITLE
Re-add deleted flaky test

### LINKED ISSUE #6817: Bring interrupting boundary event test back
**Description**

We removed the following test:

```

  @Test
  // https://github.com/camunda-cloud/zeebe/issues/6587
  public void shouldInterruptBoundaryEvent() {
    // given
    final ProcessBuilder process = Bpmn.createExecutableProcess(PROCESS_ID);

    process
        .eventSubProcess("event_sub_proc")
        .startEvent("event_sub_start")
        .interrupting(true)
        .message(b -> b.name(MSG_NAME).zeebeCorrelationKeyExpression("key"))
        .endEvent("event_sub_end");

    final BpmnModelInstance model =
        process
            .startEvent("start_proc")
            .sequenceFlowId("toParallel")
            .serviceTask("serviceTask", s -> s.zeebeJobType("task"))
            .boundaryEvent("boundary")
            .cancelActivity(true)
            .message(m -> m.name("msg").zeebeCorrelationKeyExpression("key"))
            .endEvent("end_proc")
            .done();

    engineRule.deployment().withXmlResource(model).deploy();

    final long processInstanceKey =
        engineRule
            .processInstance()
            .ofBpmnProcessId(PROCESS_ID)
            .withVariables(Map.of("key", 123, "key1", 123))
            .create();

    // when
    RecordingExporter.messageSubscriptionRecords(MessageSubscriptionIntent.CREATED)
        .withProcessInstanceKey(processInstanceKey)
        .withMessageName("msg")
        .await();

    engineRule
        .message()
        .withName("msg")
        .withCorrelationKey("123")
        .withVariables(Map.of("key", "123"))
        .publish();

    RecordingExporter.processInstanceRecords()
        .withElementId("serviceTask")
        .withProcessInstanceKey(processInstanceKey)
        .withIntent(ProcessInstanceIntent.ELEMENT_TERMINATING)
        .await();

    engineRule
        .message()
        .withName(MSG_NAME)
        .withCorrelationKey("123")
        .withVariables(Map.of("key", "123"))
        .publish();

    // then
    assertThat(
            RecordingExporter.processInstanceRecords()
                .withProcessInstanceKey(processInstanceKey)
                .limitToProcessInstanceCompleted())
        .extracting(r -> tuple(r.getValue().getBpmnElementType(), r.getIntent()))
        .containsSubsequence(
            tuple(BpmnElementType.EVENT_SUB_PROCESS, ProcessInstanceIntent.ELEMENT_ACTIVATED),
            tuple(BpmnElementType.EVENT_SUB_PROCESS, ProcessInstanceIntent.ELEMENT_COMPLETED),
            tuple(BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_COMPLETED));
  }
```

Because it was flaky. It was a regression test which we should add again, but we need to find a way how we can make sure that the boundary event is terminated. Due to the processor migration we now write the activation and complete command in a batch. This is makes it currently harder to write this test and made the test flaky.

### LINKED ISSUE #6964: React on internal Command rejections
**Description**

We introduce rejections for Commands, which can't be applied since the state is not the expect as it was when they are written. This normally happens on concurrent commands. We currently don't handle these rejections, since we assumed another flow will continue (which is not always the case)

diagram_1

Best example is this https://github.com/camunda-cloud/zeebe/issues/6817#issuecomment-832462159 where we trigger a boundary and a event sub process. Here we terminate the service task (by boundary event) and interrupt the scope (by event sub process) both follow up commands are then rejected. We can't complete the boundary event, since the scope is interrupted and we cant terminate the task, since it is already terminated. The logical consequence would be in this case to terminate the boundary event, but currently the process instance is just stuck and nothing will happen.

Maybe there is a different way to handle this, but currently for me it would make sense to just handle the rejection somehow.
Like process COMPLETE REJECTION, check whether scope is interrupted and terminate corresponding activity which will cause continuing and triggering of event sub process then.

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## Description

This test was flaky because the interruption could interfere with the completion of the boundary event. When this occurs the process instance got stuck. This has been fixed in https://github.com/camunda/zeebe/pull/11216.

## Related issues

closes #6817

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by remcowesterhoud** [comment]
Ran the test 500 times without failure. I'd say the flakiness is gone.

**Comment by korthout** [comment]
Quick note before the review: This PR should not close this issue because it attempts to discuss a larger concept, and this only tackles one instance of it.
- https://github.com/camunda/zeebe/issues/6964

> This sounds like a solution for this very specific case, though. I think it would make sense to have a convention here - either we ensure that we do not produce internal rejections (except possibly in very specific, documented cases), or we handle internal rejections. Otherwise it's unclear what the behavior is, which makes it harder to reason about the engine.

_Originally posted by @npepinpe in https://github.com/camunda/zeebe/issues/6964#issuecomment-879843799_

However, that was never discussed anymore. I'd prefer to keep that issue open, so we can actually have that discussion.

**Comment by remcowesterhoud** [comment]
Sure, I'll remove it from the closed issues

File: engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/subprocess/InterruptingEventSubprocessConcurrencyTest.java [hunk 1]
```diff
@@ -220,4 +222,65 @@ public void shouldTerminateXorWithIncident() {
...
+            .endEvent("end_proc")
+            .done();
+
+    engineRule.deployment().withXmlResource(model).deploy();
+
+    final long processInstanceKey =
+        engineRule
+            .processInstance()
+            .ofBpmnProcessId(PROCESS_ID)
+            .withVariables(Map.of("key", 123, "key1", 123))
```
Comment (korthout) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/subprocess/InterruptingEventSubprocessConcurrencyTest.java:256]: ❓ The instance is created with two vars: `key` and `key1`, but `key1` appears unused. Did you perhaps make a change to the correlation key expressions? I imagine that both were needed to differentiate the correlations or so.

File: engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/subprocess/InterruptingEventSubprocessConcurrencyTest.java [hunk 2]
```diff
@@ -220,4 +222,65 @@ public void shouldTerminateXorWithIncident() {
...
+    RecordingExporter.messageSubscriptionRecords(MessageSubscriptionIntent.CREATED)
+        .withProcessInstanceKey(processInstanceKey)
+        .withMessageName("msg")
+        .await();
+
+    engineRule.writeRecords(
+        RecordToWrite.command()
+            .message(
+                MessageIntent.PUBLISH,
+                new MessageRecord().setName("msg").setCorrelationKey("123").setTimeToLive(0)),
```
Comment (korthout) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/subprocess/InterruptingEventSubprocessConcurrencyTest.java:269]: 🔧 I think we can clarify what the tests is doing by making the message names more specific.

**Review by korthout — APPROVED** [review]
Thanks @remcowesterhoud 🧪 

👍 The changes look good to me

Please have a look at my comments before merging

File: engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/subprocess/InterruptingEventSubprocessConcurrencyTest.java (hunk 1 above)
Comment (remcowesterhoud) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/subprocess/InterruptingEventSubprocessConcurrencyTest.java:256]: I don't know, I just copied the test back in I didn't write it 😄

Comment (korthout) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/subprocess/InterruptingEventSubprocessConcurrencyTest.java:256]: Might still be a good opportunity to improve this. Perhaps just delete the second var.

**Comment by remcowesterhoud** [comment]
bors merge

**Comment by remcowesterhoud** [comment]
bors retry

**Comment by remcowesterhoud** [comment]
bors retry

**Pull request merged**